contracts/oracle/FastPriceFeed.sol

```diff
@@ -242,72 +270,140 @@ contract FastPriceFeed is ISecondaryPriceFeed, IFastPriceFeed, Governable {
+            if (_maximise) {
+                // return the _refPrice if it is the higher price
+                if (_refPrice > fastPrice) { return _refPrice; }
+                // use the maxPrice if the fastPrice exceeds the max allowed value
+                return fastPrice > maxPrice ? maxPrice : fastPrice;
+            }
 
-                if (_refPrice < fastPrice) {
-                    uint256 volPrice = fastPrice.mul(BASIS_POINTS_DIVISOR.sub(volBasisPoints)).div(BASIS_POINTS_DIVISOR);
-                    // the volPrice should not be less than _refPrice
-                    return volPrice < _refPrice ? _refPrice : volPrice;
-                }
+            // return the _refPrice if it is the lower price
+            if (_refPrice < fastPrice) { return _refPrice; }
+            // use the minPrice if the fastPrice is below the min allowed value
+            return fastPrice < minPrice ? minPrice : fastPrice;
+        }
 
-                return fastPrice;
+        if (_maximise) {
+            if (_refPrice > fastPrice) {
+                uint256 volPrice = fastPrice.mul(BASIS_POINTS_DIVISOR.add(volBasisPoints)).div(BASIS_POINTS_DIVISOR);
+                // if the _refPrice is more than the fastPrice, adjust the price upwards based on volBasisPoints
+                // from the fastPrice towards the _refPrice, the volPrice should not be more than _refPrice
+                return volPrice > _refPrice ? _refPrice : volPrice;
             }
+
+            return fastPrice;
         }
 
-        if (_maximise) {
-            if (_refPrice > fastPrice) { return _refPrice; }
-            return fastPrice > maxPrice ? maxPrice : fastPrice;
+        if (_refPrice < fastPrice) {
+            uint256 volPrice = fastPrice.mul(BASIS_POINTS_DIVISOR.sub(volBasisPoints)).div(BASIS_POINTS_DIVISOR);
+            // if the _refPrice is less than the fastPrice, adjust the price downwards based on volBasisPoints
+            // from the fastPrice towards the _refPrice, the volPrice should not be less than _refPrice
+            return volPrice < _refPrice ? _refPrice : volPrice;
         }
 
-        if (_refPrice < fastPrice) { return _refPrice; }
-        return fastPrice < minPrice ? minPrice : fastPrice;
+        return fastPrice;
     }
 
-    function favorFastPrice() public view returns (bool) {
+    function favorFastPrice(address _token) public view returns (bool) {
         if (isSpreadEnabled) {
             return false;
         }
 
         if (disableFastPriceVoteCount >= minAuthorizations) {
+            // force a spread if watchers have flagged an issue with the fast price
+            return false;
+        }
+
+        (/* uint256 prevRefPrice */, /* uint256 refTime */, uint256 cumulativeRefDelta, uint256 cumulativeFastDelta) = getPriceData(_token);
+        if (cumulativeFastDelta > cumulativeRefDelta && cumulativeFastDelta.sub(cumulativeRefDelta) > maxCumulativeDeltaDiff) {
```

Comment:
i assume it should be disabled by default? or it will achieved with very big `maxCumulativeDeltaDiff`?

Reply:
yea i think we can just start with a very large value